lib/components/position_component.dart

```diff
@@ -72,51 +78,33 @@ abstract class PositionComponent extends Component {
 
   TextConfig get debugTextConfig => TextConfig(color: debugColor, fontSize: 12);
 
-  Vector2 get position => Vector2(x, y);
-  void setPosition(Vector2 position) {
-    x = position.x;
-    y = position.y;
-  }
-
-  Vector2 get size => Vector2(width, height);
-  void setBySize(Vector2 size) {
-    width = size.x;
-    height = size.y;
-  }
-
-  /// Returns the size of this component starting at (0, 0).
-  /// Effectively this is it's position with respect to itself.
-  /// Use this if the canvas is already translated by (x, y).
-  Rect toOriginRect() => Rect.fromLTWH(0, 0, width, height);
-
   /// Returns the relative position/size of this component.
   /// Relative because it might be translated by their parents (which is not considered here).
-  Rect toRect() => Rect.fromLTWH(
-        x - anchor.relativePosition.x * width,
-        y - anchor.relativePosition.y * height,
-        width,
-        height,
-      );
+  Rect toRect() {
+    return Rect.fromLTWH(
+      x - anchor.relativePosition.x * width,
+      y - anchor.relativePosition.y * height,
+      width,
+      height,
+    );
+  }
```

Comment:
we could have an anchorPositon getter that returns `position - anchor.relativePositon.multiply(size)`
and then this method becomes `return anchorPositon.toRect(size)`

Reply:
we could even have a setter for anchorPosition that does that right math

Reply:
Much cleaner, good suggestion.